### PR TITLE
add BookDto, BookMapper and BookController classes

### PR DESCRIPTION
**Pull request by Bohadan-Markatov**

### CODE REVIEW

**Review by bodya4243 — APPROVED** [review]
all good

File: pom.xml [hunk 1]
```diff
@@ -84,6 +97,33 @@
...
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                            <version>1.18.32</version>
+                        </path>
+                        <path>
+                            <groupId>org.mapstruct</groupId>
+                            <artifactId>mapstruct-processor</artifactId>
+                            <version>1.5.5.Final</version>
```
Comment (yevheniisynytsia) [inline pom.xml]: let's create a property for versions `1.5.5.Final` and `0.2.0` to reuse them in your dependencies

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
+@AllArgsConstructor
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookshop/controller/BookController.java]:
```suggestion
@RequiredArgsConstructor

```

File: src/main/java/mate/academy/bookshop/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -36,6 +38,18 @@ public Book save(Book book) {
         }
     }
 
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> bookQuery = session.createQuery("SELECT book FROM Book book "
+                    + "WHERE book.id = :id", Book.class);
+            bookQuery.setParameter("id", id);
+            return bookQuery.uniqueResultOptional();
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookshop/repository/impl/BookRepositoryImpl.java]: could be just
`return Optional.ofNullable(session.get(Book.class, id));`

**Review by yevheniisynytsia — APPROVED** [review]
Great job! Left just some minor comments, but approved. Please take a look at the comments.

**Pull request merged**